client/src/GameState.ts

```diff
@@ -69,13 +135,26 @@ export class GameState {
             tetro.updateFromLookahead(Tetromino.rotate(tetro, i + 1));
         });
 
+        this.socket.on(
+            "reportBoard",
+            (callback: (boardState: BoardState) => void) => {
+                callback(this.toBoardState());
+            }
+        );
+
         this.socket.on("initPlayer", (playerId) => {
+            if (this.otherTetrominoes.length > 3) {
```

Comment:
This should be put into the initializePlayer function. The sceneGameArena may re-create this listener (deleting the if-condition), but if its in initializePlayer, it will still be saved.